fontforge/sfd.c

```diff
@@ -1068,6 +1080,37 @@ static void SFDDumpImage(FILE *sfd,ImageList *img) {
     fprintf(sfd,"\nEndImage\n" );
 }
 
+#ifndef _NO_LIBPNG
+static void SFDDumpImagePNG(FILE *sfd,ImageList *img) {
+    GImage *image = img->image;
+    struct _GImage *base = image->list_len==0?image->u.image:image->u.images[0];
+    struct enc85 enc;
+    int i;
+    int imsize = (int) (base->image_type==it_true?3*base->width:base->bytes_per_line);
+    char* pngbuf;
+    size_t pnglen;
+
+    FILE *pngfp = open_memstream(&pngbuf, &pnglen);
+    GImageWrite_Png(image, pngfp, false);
+    fflush(pngfp);
+
+    fprintf(sfd, "ImageX: image/png %d %d %d %x %g %g %g %g\n",
+	    (int) base->width, (int) base->height, pnglen, (int) base->trans,
```

Comment:
Is it really necessary to serialise out the base params? Surely they're already encoded as part of the png